### ISSUE TITLE
search_type error for MLP

### ISSUE THREAD

**Issue by donsmj** [keras-team/autokeras#541]
Hi all

When using Greedy searcher for MLPmodule as the following code 

MlpModule = MlpModule(loss=cross_entropy, metric=Accuracy,searcher_args={}, verbose=True,search_type=GreedySearcher)

I get a following error:

TypeError : ___init___() got an unexpected keyword argument 'search_type'

This bug must be reported and if somebody has any ideas, would be happy to share it with us.

**Comment by akg92** [comment]
Hi,
   I was actually looking into autokeras code for academic puprpose.  I am not a contributer to the project. Can't comment authentically. MLP module use BayesianSearcher now. Find the code below. If the BayesianSearcher is enough for your purpose remove the parameter.

```
`class MlpModule(NetworkModule):
    """ Class to create an MLP module."""
    def __init__(self, loss, metric, searcher_args=None, path=None, verbose=False):
        super(MlpModule, self).__init__(loss, metric, searcher_args, path, verbose)
        self.generators.extend([MlpGenerator] * 2)`
```


Init for the NetworkModule
`def __init__(self, loss, metric, searcher_args=None, path=None, verbose=False, search_type=BayesianSearcher):`

**Comment by donsmj** [comment]
Thanks. My purpose is for academia also. But my main concern is by getting the final models via default optimization method ( as you mentioned that is Baysian searcher ), the outcome of Mlpmodule fit will be 👍 


+----------------------------------------------+
|               Training model 4               |
+----------------------------------------------+
Using TensorFlow backend.

No loss decrease after 5 epochs.


Saving model.
+--------------------------------------------------------------------------+
|        Model ID        |          Loss          |      Metric Value      |
+--------------------------------------------------------------------------+
|           4            |   0.2726407766342163   |   0.8833333333333334   |
+--------------------------------------------------------------------------+


+----------------------------------------------+
|               Training model 5               |
+----------------------------------------------+
Using TensorFlow backend.

No loss decrease after 5 epochs.


Saving model.
+--------------------------------------------------------------------------+
|        Model ID        |          Loss          |      Metric Value      |
+--------------------------------------------------------------------------+
|           5            |  0.27542130947113036   |   0.8910256410256411   |
+--------------------------------------------------------------------------+


+----------------------------------------------+
|               Training model 6               |
+----------------------------------------------+
Using TensorFlow backend.

No loss decrease after 5 epochs.


Saving model.
+--------------------------------------------------------------------------+
|        Model ID        |          Loss          |      Metric Value      |
+--------------------------------------------------------------------------+
|           6            |  0.27543354630470274   |   0.8782051282051281   |
+--------------------------------------------------------------------------+


+----------------------------------------------+
|               Training model 7               |
+----------------------------------------------+
Using TensorFlow backend.
Time is out.

No loss decrease after 5 epochs.


Then I put this code as the final_fit and nothing appears. I also trained it with several arguments and noting appeared,

 
mlpModule.final_fit(train_data, test_data, trainer_args={'max_iter_num': 6}, retrain=False)
